### PR TITLE
Add automated testing and fix bug in question date. 

### PR DESCRIPTION
**Pull request by aditya102**
### Add automated testing for views and Published Question
1. Check for no Questions.
    * If no questions exits there should be an appropriate message.

2. Check for question with past date.
    * Question with past date should be displayed on the index page.

3. Check for question with future date.
     * Question with future date should not be on the index page.

4. Fixed Bug which shows the question with future date in current day.

### CODE REVIEW

File: polls/models.py [hunk 1]
```diff
@@ -12,7 +12,8 @@ def __str__(self):
         return self.question_text
 
     def was_published_recently(self):
-        return self.pub_date >= timezone.now() - datetime.timedelta(days=1)
+        now = timezone.now()
+        return now - datetime.timedelta(days=1) <=self.pub_date <=now
```
Comment (waranlogesh) [inline polls/models.py]:
```suggestion
        return now - datetime.timedelta(days=1) <= self.pub_date <= now
```
Please use flake8 proceeding forward to check your code. Google on what flake8 is and how it should be used.

File: polls/tests.py [hunk 2]
```diff
@@ -1,2 +1,54 @@
+import datetime
+
 from django.test import TestCase
+from django.utils import timezone
+from django.urls import reverse
+
+from .models import Question, Choice
+
+def create_question(question_text, days):
```
Comment (waranlogesh) [inline polls/tests.py:10]:
```suggestion

def create_question(question_text, days):
```
Two line breaks to separate outer class/method.

File: polls/tests.py [hunk 3]
```diff
@@ -1,2 +1,54 @@
...
+    def test_past_question(self):
+        create_question(question_text="Past question.", days=-30)
+        response = self.client.get(reverse('polls:index'))
+        self.assertQuerysetEqual(
+            response.context['latest_question_list'],
+            ['<Question: Past question.>']
+        )
+
+    def test_future_question(self):
+
```
Comment (waranlogesh) [inline polls/tests.py]:
```suggestion
```

File: polls/tests.py [hunk 4]
```diff
@@ -1,2 +1,54 @@
...
+    def test_future_question(self):
+    
+        create_question(question_text="Future question.", days=30)
+        response = self.client.get(reverse('polls:index'))
+        self.assertContains(response, "No polls are available.")
+        self.assertQuerysetEqual(response.context['latest_question_list'], [])
+
+    def test_future_question_and_past_question(self):
+
```
Comment (waranlogesh) [inline polls/tests.py]:
```suggestion
```

File: polls/tests.py [hunk 5]
```diff
@@ -1,2 +1,54 @@
...
+        create_question(question_text="Future question.", days=30)
+        response = self.client.get(reverse('polls:index'))
+        self.assertQuerysetEqual(
+            response.context['latest_question_list'],
+            ['<Question: Past question.>']
+        )
+
+    def test_two_past_questions(self):
 
+        create_question(question_text="Past question 1.", days=-30)
```
Comment (waranlogesh) [inline polls/tests.py:50]: Remove the above line break.

File: polls/tests.py [hunk 6]
```diff
@@ -1,2 +1,54 @@
...
+        response = self.client.get(reverse('polls:index'))
+        self.assertContains(response, "No polls are available.")
+        self.assertQuerysetEqual(response.context['latest_question_list'], [])
+
+    def test_future_question_and_past_question(self):
+       
+        create_question(question_text="Past question.", days=-30)
+        create_question(question_text="Future question.", days=30)
+        response = self.client.get(reverse('polls:index'))
+        self.assertQuerysetEqual(
```
Comment (waranlogesh) [inline polls/tests.py:44]: Line break to separate asserts in all test cases.

**Review by waranlogesh — CHANGES_REQUESTED** [review]

**Comment by waranlogesh** [comment]
PR title should be Add unit tests and fix a bug in question date

File: polls/tests.py [hunk 7]
```diff
@@ -1,2 +1,55 @@
...
+    def test_past_question(self):
+        create_question(question_text="Past question.", days=-30)
+        response = self.client.get(reverse('polls:index'))
+        self.assertQuerysetEqual(
+            response.context['latest_question_list'],
+            ['<Question: Past question.>']
+        )
+
+    def test_future_question(self):
+
```
Comment (waranlogesh) [inline polls/tests.py]: Remove this line break.

**Review by waranlogesh — CHANGES_REQUESTED** [review]

File: polls/tests.py (hunk 2 above)
Comment (aditya102) [reply polls/tests.py:10]: Made the change

**Pull request merged**